File: app/src/main/java/com/thecoolguy/rumaan/fileio/data/db/UploadItemDao.java
package com.thecoolguy.rumaan.fileio.data.db;

import static com.thecoolguy.rumaan.fileio.data.db.DatabaseContract.TABLE_NAME;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import com.thecoolguy.rumaan.fileio.data.models.FileEntity;
import java.util.List;

@Dao
public interface UploadItemDao {

  @Insert
  long insert(FileEntity fileEntity);

  @Query("SELECT DISTINCT * FROM " + TABLE_NAME + " where id = :id")
  FileEntity getItem(long id);

  @Delete
  void delete(FileEntity... fileEntity);

  @Query("DELETE FROM " + TABLE_NAME)
  void clearAll();

  @Query("SELECT DISTINCT count(*) FROM " + TABLE_NAME)
  int numberOfItems();

  @Query("SELECT * FROM " + TABLE_NAME)
  LiveData<List<FileEntity>> getAllUploads();

}

File: app/src/main/java/com/thecoolguy/rumaan/fileio/data/db/UploadItemDao.kt
package com.thecoolguy.rumaan.fileio.data.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.thecoolguy.rumaan.fileio.data.models.FileEntity

@Dao
interface UploadItemDao {
    @get:Query("SELECT * FROM " + DatabaseContract.TABLE_NAME)
    val allUploads: LiveData<List<FileEntity>>

    @Insert
    fun insert(fileEntity: FileEntity): Long

    @Query("SELECT DISTINCT * FROM " + DatabaseContract.TABLE_NAME + " where id = :id")
    fun getItem(id: Long): FileEntity

    @Query("DELETE FROM ${DatabaseContract.TABLE_NAME}" + " WHERE id = :id")
    fun deleteItemWithId(id: Long)

    @Delete
    fun delete(vararg fileEntity: FileEntity)

    @Query("DELETE FROM " + DatabaseContract.TABLE_NAME)
    fun clearAll()

    @Query("SELECT DISTINCT count(*) FROM " + DatabaseContract.TABLE_NAME)
    fun numberOfItems(): Int

}
